Machine_Learning/notebooks/czmodel/Regresssion_PyTorch_5_0_0.ipynb

```diff
@@ -0,0 +1,669 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "hrIPBmrkrEEG"
+   },
+   "outputs": [],
+   "source": [
+    "from torch.utils.data import DataLoader, Dataset\n",
+    "import cv2\n",
+    "\n",
+    "# Dataset class\n",
+    "class SampleDataset(Dataset):\n",
+    "    def __init__(self, images_folder:str, labels_folder:str, transforms=None):\n",
+    "        self.sample_images = sorted([os.path.join(images_folder, f) for f in os.listdir(images_folder) \n",
+    "                                    if os.path.isfile(os.path.join(images_folder, f))])\n",
+    "        self.sample_labels = sorted([os.path.join(labels_folder, f) for f in os.listdir(labels_folder) \n",
+    "                                    if os.path.isfile(os.path.join(labels_folder, f))])\n",
+    "        self.transforms = transforms\n",
+    "\n",
+    "    def __getitem__(self, idx):\n",
+    "        image = self.sample_images[idx]\n",
+    "        label = self.sample_labels[idx]\n",
+    "        img = cv2.imread(image, cv2.IMREAD_GRAYSCALE)\n",
+    "        labl = cv2.imread(label, cv2.IMREAD_GRAYSCALE) \n",
+    "\n",
+    "        if self.transforms is not None:\n",
+    "            img = self.transforms(img)\n",
+    "            labl = self.transforms(labl)\n",
+    "\n",
+    "        return img.to(torch.float32), labl.to(torch.float32)\n",
+    "\n",
+    "    def __len__(self):\n",
+    "        return len(self.sample_images)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define transforms\n",
+    "from torchvision import transforms\n",
+    "transforms = transforms.Compose([\n",
+    "    transforms.ToTensor(),\n",
+    "    transforms.CenterCrop(CROP_SIZE),\n",
+    "])\n",
+    "\n",
+    "# Create dataset and dataloader\n",
+    "sample_dataset = SampleDataset(images_folder=IMAGES_FOLDER,\n",
+    "                               labels_folder=LABELS_FOLDER,\n",
+    "                               transforms=transforms)\n",
+    "\n",
+    "sample_dataloader = DataLoader(dataset=sample_dataset,\n",
+    "                                      batch_size=2,\n",
+    "                                      shuffle=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "ZRujNv7nrEEG"
+   },
+   "source": [
+    "### Define a simple model\n",
+    "This part defines a simple PyTorch convolutional model for regression task."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "prtTh7t2ZubC"
+   },
+   "outputs": [],
+   "source": [
+    "# Define simple PyTorch convolutional model\n",
+    "class SampleModel(nn.Module):\n",
+    "\n",
+    "    def __init__(self):\n",
+    "        super(SampleModel, self).__init__()\n",
+    "        self.conv = nn.Sequential(\n",
```

Comment:
Sorry to be nitpicking! I would rather call this `model` or something similar since `conv` might misleadingly suggest that it's a single convolutional layer.